src/App.tsx

```diff
@@ -1,14 +1,73 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
+function getFilteredTodos(todos: Todo[], query: string, queryInput: string) {
+  const lowerCaseQueryInput = queryInput.toLowerCase();
+  const preparedTodos = todos.filter(todo => {
+    if (query === 'active') {
+      return todo.completed === false;
+    } else if (query === 'completed') {
+      return todo.completed === true;
+    } else {
+      return todos;
+    }
+  });
+
+  let readyTodos;
+
+  if (queryInput !== '') {
+    readyTodos = preparedTodos.filter(readyTodo =>
+      readyTodo.title.toLowerCase().includes(lowerCaseQueryInput),
+    );
+
+    return readyTodos;
+  }
+
+  return preparedTodos;
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
```

Comment:
Rename all boolean variables using [tips](https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig)

```suggestion
  const [isLoading, setIsLoading] = useState(false);
```